### PR TITLE
Include Remote in the class names of provider, backends and jobs

### PR DESCRIPTION
**Pull request by MaxleDrut**
We recently introduced a version of the provider handling the local simulation of cat qubits. These local simulations can be accessed using the `AliceBobLocalProvider`. 

To make sure users understand the difference between the local provider and the remote one, targeting the Alice & Bob endpoints, this PR  renames `AliceBobProvider`, `AliceBobBackends` and `AliceBobJobs` to include `Remote` so we can better differentiate them from local.

These changes were tested, including with the integration tests. I also made sure the README provided code was properly working.

### CODE REVIEW

File: qiskit_alice_bob_provider/backend.py [hunk 1]
```diff
@@ -24,17 +24,16 @@
...
+from .job import AliceBobRemoteJob
 from .qir_to_qiskit import ab_target_to_qiskit_target
 from .utils import camel_to_snake_case, snake_to_camel_case
 
 
-class AliceBobBackend(BackendV2):
-    """Class representing the single cat simulator target accessible via the
-    Alice & Bob API"""
+class AliceBobRemoteBackend(BackendV2):
+    """Class representing the targets accessible via the Alice & Bob API."""
```
Comment (MaxleDrut) [inline qiskit_alice_bob_provider/backend.py:33]: Updating the docstring in a Driveby. The targets also would include multiple cats and actual hardwares, not only simulators.

Comment (jrj-d) [reply qiskit_alice_bob_provider/backend.py:33]: Great thanks!

**Review by jrj-d — APPROVED** [review]
LGTM
